src/server/middleware/saveStoryFile.js

```diff
@@ -55,8 +48,8 @@ fluid.defaults("sjrk.storyTelling.server.middleware.saveStoryWithBinaries", {
  * @param {Object} file - an uploaded file to process
  * @param {filenameResolverCallback} cb - the callback
  */
-sjrk.storyTelling.server.middleware.saveStoryWithBinaries.filenameResolver = function (req, file, cb) {
-    var id = uuidv1();
+sjrk.storyTelling.server.middleware.saveStoryFile.filenameResolver = function (req, file, cb) {
+    var id = uuidv4();
```

Comment:
Indeed, I was thinking of passing in model detail containing the current filename in order to remove that from the server before operating on the new file. Since this is multi-part form data, it should be feasible.

I'd also considered creating the ID client-side, but decided to keep all ID generation on the server.